### DISCUSSION TITLE
dompdf freezes at $dompdf->render();

### DISCUSSION THREAD

**Discussion by ballotade** [dompdf/dompdf#3299] [Q&A]
Hello,

Suddenly my Programm freezes at $dompdf->render();
a few weeks ago it worked very well.
I habe installed dompdf 2.0.3 and xampp with PHP Version 8.2.4.

Some ideas what I can try to get it work?

Thanks
Greetings from Germany

**Comment by bsweeney** [comment]
Dompdf is probably not frozen. Typically this occurs when dompdf is stuck in a loop attempting to page an element. You'll need to share an example of your HTML+CSS before we could provide much guidance.

**Comment by ballotade** [comment]
Hey bsweeney,
thanks for your reply.

Here is an example of the html, that I want dompdf to create a pdf from.
I took the base64-data from the images away.

```
<!doctype html><html><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
    
<style>
    * {
	   -webkit-box-sizing: border-box;
	       -moz-box-sizing: border-box;
		       box-sizing: border-box;
    }
    
    /*html, body { margin: 0; }*/
    .seitenUmbruch {
        page-break-before: always;
    }
    @page { margin: 0; }
    @font-face {
    font-family: ARDARLING;
    src: url("../publichtml/user/11//ARDARLING.ttf");
    font-weight: normal;
    font-style: normal;
}</style>
</head>
<body>
<div class='' style='position: absolute; top: 0; left: 0; width: 815pt; height; 1110pt;  font-family: ARDARLING;  background-color: #0000f0;'>
<div style='position: absolute; top: 0pt; left: 0pt; width: 815pt; height:1110pt; opacity: 1; background-size: contain; background-position: 50% 50%; background-repeat: no-repeat; background-image: url(data:image/png;base64,…);'></div>
<div style='position: absolute; width: 260pt; height: 260pt; top: 83pt; left: 484pt; text-align: left; font-size: 94pt; color: #000000; overflow: hidden;'>
<div style='position: absolute; width: 260pt; height: 260pt; top: 0%; left: 0%; background-position: 50% 50%; background-size: contain; background-repeat: no-repeat; background-image: url(data:image/png;base64,…'></div>
</div><div style='position: absolute; width: 236pt; height: 343pt; top: 0pt; left: 71pt; text-align: left; font-size: 295pt; color: #000000; overflow: hidden;'>
<div style='position: relative; padding: 24pt 24pt 24pt 24pt; line-height: 295pt; font-size: 295pt'>7</div>
</div>
<div style='position: absolute; width: 815pt; height: 1110pt; top: 0pt; left: 0pt; text-align: left; font-size: 94pt; color: #000000; overflow: hidden;'>
<div style='position: absolute; width: 872pt; height: 1188pt; top: 2%; left: -12%; background-position: 50% 50%; background-size: contain; background-repeat: no-repeat; background-image: url(data:image/png;base64,…);'></div>
</div>
</div></body>
```

**Reply by bsweeney** [reply]
I don't see anything here that would directly cause an issue (I tested in the [debug helper](https://eclecticgeek.com/dompdf/debug.php)). What is the paper size? And DPI (which can impact background image rendering performance)?

**Comment by ballotade** [comment]
thanks for your help @bsweeney 

I design card games, so it is not very large. An I only try to export one card and it never finishes...
Here is my code for the size:

`$mmtopixel=300/25.4;
$border=3;
$px=round((63+$border*2)*$mmtopixel);
$py=round((88+$border*2)*$mmtopixel);
$customPaper = array(0,0,$px,$py);
`

**Reply by bsweeney** [reply]
My initial impression is that an element is being infinitely paged. But I don't see anything in what you provided that would cause that type of issue. I used both large (a2) and small (a8) paper sizes without issue.

Is your HTML content loaded over HTTP, local filesystem, or just a runtime-generated string? Are your images fairly large?

**Reply by ballotade** [reply]
The images are png-files and with a size about 200kB.

It is loaded over local file system on my notebook with xampp.
It worked when I tried last time a year ago. I worked with php version 7.x. Now with 8.2.  Could this be the problem?

**Reply by ballotade** [reply]
update: I tried to create a pdf without the images and it worked...

**Reply by bsweeney** [reply]
What type of images are they? There are certain formats that have proven problematic in the past. can you share the images?

**Comment by ballotade** [comment]
![gelb50](https://github.com/dompdf/dompdf/assets/109868857/3bff6a96-9466-4cdd-a2f7-7d8baf7f4973)

Here is an example of a background-image

**Reply by ballotade** [reply]
Update: i tried to render only this background image and after a long time it worked...

**Reply by bsweeney** [reply]
Rendering that image withing your sample HTML and the Dompdf DPI set to 72 it was fast enough for me (~1.5s). Rendering with a DPI of 300 was definitely slower but not to the point where I would consider Dompdf frozen (~18s).

**Comment by ballotade** [comment]
Hello bsweeney,
I installed an older version of xampp (php 8.0.28) and then it throwed the following error:
Fatal error: Uncaught Exception: The PHP GD extension is required, but is not installed. in C:\xampp\htdocs\SpieleMacher\src\vendor\dompdf\dompdf\src\Renderer\AbstractRenderer.php:284 Stack trace: #0 [...]

So I opened the php.ini file and deleted the semicolon before 
extension=gd
and now it works.
Thank you for your help. I think it was the same Problem with PHP 8.3. but I didnt get the error message.

**Reply by bsweeney** [reply]
Oh, OK then. It may have been the logging level that prevented PHP 8.2 from reporting the error. I should have thought to suggest [enabling additional debug messaging](https://github.com/dompdf/dompdf/wiki/Debugging-Dompdf).

Glad you were able to figure it out.
